libs/feature/search/src/lib/router/state/router.effects.ts

```diff
@@ -0,0 +1,70 @@
+import { Location } from '@angular/common'
+import { Injectable } from '@angular/core'
+import { ActivatedRouteSnapshot, Router } from '@angular/router'
+import { Actions, createEffect, ofType } from '@ngrx/effects'
+import { tap } from 'rxjs/operators'
+
+import * as RouterActions from './router.actions'
+import * as MdViewActions from '../../metadata-view/state/mdview.actions'
+import { navigation } from '@nrwl/angular'
+import { MetadataRouteComponent, SearchRouteComponent } from '../constants'
+
+@Injectable()
+export class RouterEffects {
+  constructor(
+    private _actions$: Actions,
+    private _router: Router,
+    private _location: Location
+  ) {}
+
+  navigate$ = createEffect(
+    () =>
+      this._actions$.pipe(
+        ofType(RouterActions.goAction),
+        tap(({ path, query: queryParams }) => {
+          this._router.navigate([path], { queryParams })
+        })
+      ),
+    { dispatch: false }
+  )
+
+  loadMetadata$ = createEffect(() =>
+    this._actions$.pipe(
+      navigation(MetadataRouteComponent, {
```

Comment:
this comes from Nrwl, actually, the router state management took from carto-ui was hand made, maybe nrwl had more recent/clean implementation of it somewhere.

Reply:
yeah I figured I might as well not write a custom serializer and rely on nrwl helpers.